File: Tree/BinaryTreeLevelorderTraversalII/BinaryTreeLevelorderTraversalII.java
/*
class TreeNode {
	int val;
	TreeNode left, right;
	TreeNode(int v) {
		this.left = null;
		this.right = null;
		this.val = v
	}
}
*/
class Solution {
	public list<List<Integer>> bottomLevelOrderTraverse(TreeNode root) {
		List<List<Integer>> result =new ArrayList<List<Integer>>();
		if(root == NULL)
			return result;
		
		stack<ArrayList<Integer>> s = new stack<ArrayList<Integer>>();
		
		Queue<TreeNode> q = new LinkedList<TreeNode>();
		q.offer(root);
		while(!q.isEmpty()) {
			int qLen = q.size();
			List<Integer> list = new ArrayList<>();
			
			for(int i = 0; i < qLen; ++i) {
				TreeNode node = q.poll();
				list.add(node.val);
				if(node.left)
					q.offer(node.left);
				if(node.right)
					q.offer(node.right);
			}
			
			s.push(new ArrayList<Interger>(list));
		}
		while(!s.isEmpty())
			result.add(s.pop());
		return result;
	}
}

File: Tree/BinaryTreeLevelorderTraversalII/BinaryTreeLevelorderTraversalII.cc
/*
class TreeNode {
	int val;
	TreeNode *left, *right;
	TreeNode(int v) : val(v), left(NULL), right(NULL) {}
};
*/

class Solution {
	public:
		vector<vector<int>> levelOrderTraversal(TreeNode* root) {
			vecotr<vector<int>> result;
			
			if(root == NULL);
				return result;
				
			stack<vector<int>> s;	
			queue<TreeNode*> q;
			q.push(root);
			while(!q.empty()) {
				vector<int> list;
				int size = q.size();
				for(int i = 0; i < size; ++i) {
					TreeNode* node = q.front();
					q.pop();
					list.push_back(node->val);
					if(node->left)
						q.push(node->left);
					if(node->right)
						q.push(node->right);
						
				}
				
				s.push_back(vector<int>(list));				
			}
			while(!s.isEmpty())
				result.push(s.pop());
			return result;
		}
};